### DISCUSSION TITLE
[Files Service] How to save file with the actual name?

### DISCUSSION THREAD

**Discussion by flower-of-the-bridges** [mia-platform/community#139] [Q&A] [answered]
Hi!

I would like to save a file on a S3 bucket using the `files service`, but keeping the original file name. 

Reading [the official doc](https://docs.mia-platform.eu/docs/runtime_suite/files-service/configuration), it seems that the service creates a unique identifier for each file uploaded. 

However, in some cases, it would be useful to search on a bucket by using the file name that was chosen at the moment of the upload. Is that possible or a feature that could be added in future versions of the `files service`?

**Comment by fredmaggiowski** [comment]
Using the original filename could create collision issues, and unexpected file overrides due to different files with the same name being uploaded multiple times. This is why the files service does not implement such feature. Do you have a specific use case where this is necessary?

Consider that you can still download the file with the original name using the `downloadWithOriginalName` or the `useOriginalName` query parameters 

https://docs.mia-platform.eu/docs/runtime_suite/files-service/usage#download-get-downloadfile

**Reply by flower-of-the-bridges** [reply]
The use case that I'm currently facing is where there exists end users that would like to see pictures from the hierarchical structure of the S3 bucket provided by the AWS console.

So in that particular cases they are not using the `GET` methods provided by the files service, unfortunately.

**Reply by fredmaggiowski** [reply]
As of now I don't think the files service has support for providing a path with the filename, does it? If so it may be supported but there would still be collision issues.

**Reply by fredmaggiowski** [reply]
Any update on this matter?

**Reply by flower-of-the-bridges** [reply, marked as answer]
Hi @fredmaggiowski 

At the moment we need to access the data directly on the bucket (so without the files service `GET` method) following the same naming convention provided by the client during the request.

If it's a feature that can be useful for the plugin, I'll try to add it in the road-map, otherwise I'll send the data directly on the bucket using a custom service

Thanks anyway!
